### ISSUE TITLE
Disable the accessibility outline for non-keyboard uses

### ISSUE THREAD

**Issue by ianstormtaylor** [chakra-ui/chakra-ui#3449]
Right now Chakra adds an outline (actually a `box-shadow`) to elements that receive focus, even when they are focused by clicking the mouse. For example:

<img width="623" alt="image" src="https://user-images.githubusercontent.com/311752/109063986-d3d55b80-76b7-11eb-8bcf-1f4dc3283f6b.png">

This is to aid in keyboard accessibility, but it shouldn't happen when navigating via a mouse—check how popular websites handle this, including GitHub.

It looks like this was somehow integrated in `1.0` but the solution in https://github.com/chakra-ui/chakra-ui/issues/2016#issuecomment-691674038 leads to a broken link.

This behavior has been brought up before (eg. https://github.com/chakra-ui/chakra-ui/issues/2016, https://github.com/chakra-ui/chakra-ui/issues/708, https://github.com/chakra-ui/chakra-ui/issues/574) and the response has been to read [this Medium article](https://medium.com/@keeganfamouss/accessibility-on-demand-with-chakra-ui-and-focus-visible-19413b1bc6f9) and implement it. But this is something that 100% of users would need to do to, so it should be fixed in Chakra instead by default.

By keeping this broken default Chakra is actually encouraging devs to solve it incorrectly, using something like `box-shadow: none !important` regardless of navigation style. (eg. https://github.com/chakra-ui/chakra-ui/issues/708, https://github.com/chakra-ui/chakra-ui/issues/668)

The current default is actually going to result in worse accessibility.

**Comment by spykr** [comment]
I agree that this note which can only be found in the "Upgrading to v1" page should be made much more visible in the documentation:

![image](https://user-images.githubusercontent.com/890171/109103658-65f46700-777f-11eb-8f89-870047f408c6.png)

However I disagree that "100% of users would need to do [this]", I think it's purely a personal preference. Hiding focus outlines still makes your site less accessible even if it's only for non-keyboard users.

**Comment by ianstormtaylor** [comment]
@spykr thank you! I was trying to figure out where that was documented.

> However I disagree that "100% of users would need to do [this]", I think it's purely a personal preference. Hiding focus outlines still makes your site less accessible even if it's only for non-keyboard users.

I don't think that's a good idea for two reasons:

1. Based on browsing pretty much every popular site, no one does what Chakra does with outlines for non-keyboard navigation. So it's clearly not a popular choice, and those sites have taken accessibility into account.

2. Based on the existing issues in this repo, the "fix" people will use for this is to disable the outlines entirely, way worse for accessibility. (You can try to fight this with documentation, but many won't find it and will just use the nuclear option.)

I really do think that the existing default Chakra has chosen actually hurts accessibility more than helps it.

And it looks like the [`:focus-visible` selector is supported in Chrome, Edge, and Firefox now](https://caniuse.com/css-focus-visible) so this is something that should be baked into Chakra by default, instead of requiring an extra package. (The external package is still a good idea for polyfilling non-supporting browsers like Safari.)

**Comment by ianstormtaylor** [comment]
^ I don’t think this bot is configured correctly to make Chakra better.

**Comment by lowercase00** [comment]
Any news on this topic?

**Comment by Ploppy3** [comment]
I just run into an issue using the proposed `focus-visible` solution:

Modals lose their shadows when they lose focus, which probably passed under the radar because modals are initially focused when opened.

Any solution to that?

**Comment by santialbo** [comment]
Please don't mark this as stale

**Comment by basicserge** [comment]
Also interested in this solution. Vote for that! 

As for me, it would be more or less acceptable if CSSReset included rules not only for [data-focus-visible-added], but for the class as well. I can add a data attribute to some components, but the others are overwriten. In such a case using :focus:not(.focus-visible) could do the job. 

What would be a workaround for now? Is it a good idea to hardcode a customized CSSReset?

**Comment by slava-lu** [comment]
Just wonder why this ugly thing is called an accessibility feature.  How come that a person can used a keyboard but can't use a mouse. He needs an arm for any of those. 
Thanks for not putting a mandatory Covid vaccination banner in your default theme.

**Comment by sgehrman** [comment]
same issue.

**Comment by mix3d** [comment]
I'm having issues with the focus system as well. Is there a way to force the `data-focus-visible-added` value on a component? I'm trying to manually specify it for a specific component (so that I can CHOOSE to show the focus in a specific scenario), but it seems when using the popper trigger, it overwrites the props and removes my manually specified `data-focus-visible-added` attribute.

Checked the issues first and found this, so my above issue MIGHT be tangential, but it's definitely related to this topic.
